### PR TITLE
Cleanup zarr directory representation

### PR DESCRIPTION
**Pull request by jstriebel**
This PR
* simplifies the zarr directory html pages
* makes them compatible with python fsspec
* and ensures that the pages are used in the browser always with a trailing slash 

### URL of deployed dev instance (used for testing):
- https://htmldirectoryfsspeccompat.webknossos.xyz

### Steps to test:
I tested those myself, review should be enough:
- Go to https://htmldirectoryfsspeccompat.webknossos.xyz/data/zarr/sample_organization/ROI2017_wkw?token=…,
  and explore some folders (token needs to be re-added manually)
- For the python client:
```python
from upath import UPath

p = UPath(
    "https://htmldirectoryfsspeccompat.webknossos.xyz/data/zarr/sample_organization/ROI2017_wkw",
    client_kwargs={"headers": {"X-Auth-Token": "…"}}
)

p.is_dir()
# True

list(p.iterdir())
# [
#   HTTPPath('https://htmldirectoryfsspeccompat.webknossos.xyz/data/zarr/sample_organization/ROI2017_wkw/color'),
#   HTTPPath('https://htmldirectoryfsspeccompat.webknossos.xyz/data/zarr/sample_organization/ROI2017_wkw/segmentation')
# ]

list(p.iterdir())[0].is_dir()
# True
```

### Issues:
- fixes #

------

## TODO
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

**Comment by normanrz** [comment]
I piggy-backed on this PR to change some styling.

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala [hunk 1]
```diff
@@ -73,9 +71,7 @@ class ZarrStreamingController @Inject()(
             views.html.datastoreZarrDatasourceDir(
               "Datastore",
               "%s/%s/%s".format(organizationName, dataSetName, dataLayerName),
-              mags.map { mag =>
-                (mag.toMagLiteral(), s"$dataLayerName/${mag.toMagLiteral()}")
-              }.toMap
+              mags.map(mag => mag.toMagLiteral())
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala:74]: if I remember correctly, sth. like `mags.map(_.toMagLiteral())` would work as well, right? Just wondering, I don't really care which version should be used.

Comment (fm3) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala:74]: I’d prefer the shorter version. Also the empty call parentheses can be omitted

**Review by fm3 — APPROVED** [review]
LGTM :) note my small comment on code style

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala (hunk 1 above)
Comment (jstriebel) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala:74]:
```suggestion
              mags.map(_.toMagLiteral)
```

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingController.scala [hunk 2]
```diff
@@ -254,9 +254,7 @@ class VolumeTracingController @Inject()(
             views.html.datastoreZarrDatasourceDir(
               "Tracingstore",
               "%s".format(tracingId),
-              existingMags.map { mag =>
-                (mag.toMagLiteral(allowScalar = true), mag.toMagLiteral(allowScalar = true))
-              }.toMap
+              existingMags.map(mag => mag.toMagLiteral(allowScalar = true))
```
Comment (jstriebel) [inline webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/controllers/VolumeTracingController.scala:257]:
```suggestion
              existingMags.map(_.toMagLiteral(allowScalar = true))
```

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala (hunk 1 above)
Comment (fm3) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala]: oops, CI disagrees about the parentheses… my bad, sorry!

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala [hunk 3]
```diff
@@ -71,7 +71,7 @@ class ZarrStreamingController @Inject()(
             views.html.datastoreZarrDatasourceDir(
               "Datastore",
               "%s/%s/%s".format(organizationName, dataSetName, dataLayerName),
-              mags.map(mag => mag.toMagLiteral())
+              mags.map(_.toMagLiteral)
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala:74]: @fm3 This does not seem to work, can you help me with the syntax here?

Comment (jstriebel) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala:74]:
```suggestion
              mags.map(_.toMagLiteral())
```

**Pull request merged**